cmake/CheckMemorySanitizer.cmake

```diff
@@ -0,0 +1,16 @@
+include_guard(GLOBAL)
+include(CheckCSourceCompiles)
+
+function(check_memory_sanitizer output)
+  set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+  check_c_source_compiles("
+    #if defined(__has_feature)
+    #  if __has_feature(memory_sanitizer)
+    //   MemorySanitizer is enabled.
```

Comment:
nit: We should probably use the same test program on configure and CMake (and perhaps avoid `//` comments which are invalid in C89, though that really is more of a philosophical convern.)

Reply:
> nit: We should probably use the same test program on configure and CMake

I'd prefer align Autotools' `SECP_MSAN_CHECK` with CMake's `check_memory_sanitizer` to keep the latter straightforward without negated values.

> ... and perhaps avoid `//` comments which are invalid in C89

Thanks. [Fixed](https://github.com/bitcoin-core/secp256k1/pull/1532#issuecomment-2133405610).

Reply:
> I'd prefer to align Autotools' `SECP_MSAN_CHECK` code with CMake's `check_memory_sanitizer` to keep the latter straightforward without negated values.

Fine with that if you want to add a commit that makes them consistent, but then we'll also need to `#error` if `!defined(__has_feature)`.

Reply:
> ... but then we'll also need to `#error` if `!defined(__has_feature)`.

Thanks! [Fixed](https://github.com/bitcoin-core/secp256k1/pull/1532#issuecomment-2134674954).